Sources/NIOSSL/SSLContext.swift

```diff
@@ -472,6 +485,32 @@ extension NIOSSLContext {
         if result == 0 {
             let errorStack = BoringSSLError.buildErrorStack()
             throw BoringSSLError.unknownError(errorStack)
+        } else if sendCANames, !isDirectory {
+            // For single CA file, add the CA name from the trust root.
+            // This could be from a location like /etc/ssl/cert.pem as an example.
+            CNIOBoringSSL_SSL_CTX_set_client_CA_list(context, CNIOBoringSSL_SSL_load_client_CA_file(path))
+        } else if sendCANames, isDirectory {
+            // If the path that is passed in is a directory, scan the directory and gather up the PEM files.
+            var pemFilePaths: [String] = []
+            if let dir = opendir(path) {
+                while let dirent: UnsafeMutablePointer<dirent> = readdir(dir) {
+                    let fileName = withUnsafePointer(to: &dirent.pointee.d_name) { (ptr) -> String in
+                        return ptr.withMemoryRebound(to: CChar.self, capacity: Int(ptr.pointee.0) + 1) { String(cString: $0) }
+                    }
+                    if fileName.suffix(4) == ".pem" {
+                        pemFilePaths.append(path + fileName)
+                    }
+                }
+                closedir(dir)
+            }
```

Comment:
We need to factor this code out into its own method for more easy auditing.

Reply:
Actually, maybe we should just refactor the directory iteration code: the sequence of calls to `opendir`, `readdir`, and `closedir`. This seems like a great candidate to be a custom `Sequence` whose element type is `String`. Then we can simply express `pemFilePaths` as `pemFilePaths = DirectoryContents(path).filter { $0.suffix(4) == ".pem" }`.

Reply:
Note that there's also no good reason to constrain ourselves only to `PEM`. Instead, we should probably just try both `pem` and `der` for a given cert, probably hinted by (but not limited by) the filename.

Reply:
Added in the latest commit.